modules/launcher/src/main/java/org/mule/module/launcher/ArtifactDeploymentListenerAdapter.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.launcher;
+
+
+import static java.lang.String.format;
+import static org.mule.config.bootstrap.ArtifactType.ALL;
+import static org.mule.config.bootstrap.ArtifactType.APP;
+import static org.mule.config.bootstrap.ArtifactType.DOMAIN;
+import org.mule.api.MuleContext;
+import org.mule.config.bootstrap.ArtifactType;
+
+/**
+ * Adapts an {@link ArtifactDeploymentListener} to work as a {@link DeploymentListener}.
+ */
+public class ArtifactDeploymentListenerAdapter
+{
+
+    private DeploymentListener adaptedDomainDeploymentListener;
+    private DeploymentListener adaptedApplicationDeploymentListener;
+
+    public ArtifactDeploymentListenerAdapter(ArtifactDeploymentListener artifactDeploymentListener)
+    {
+        adaptedDomainDeploymentListener = new AdaptedDeploymentListener(artifactDeploymentListener, DOMAIN);
+        adaptedApplicationDeploymentListener = new AdaptedDeploymentListener(artifactDeploymentListener, APP);
+    }
+
+    public DeploymentListener getDomainDeploymentListener()
+    {
+        return adaptedDomainDeploymentListener;
+    }
+
+    public DeploymentListener getApplicationDeploymentListener()
+    {
+        return adaptedApplicationDeploymentListener;
+    }
+
+    static class AdaptedDeploymentListener implements DeploymentListener
+    {
+
+        public static final String UNSUPPORTED_ARTIFACT_TYPE_ERROR = format("ArtifactListeners only supports %s and %s artifact types.", APP, DOMAIN);
```

Comment:
Instead of listing each supported type, use a single placeholder receiving a collection